rustls/src/msgs/handshake.rs

```diff
@@ -548,6 +528,32 @@ impl TlsListElement for CertificateCompressionAlgorithm {
     const SIZE_LEN: ListLength = ListLength::U8;
 }
 
+/// A precursor to `ClientExtensions`, allowing customisation.
+///
+/// This is smaller than `ClientExtensions`, as it only contains the extensions
+/// we need to vary between different protocols (eg, TCP-TLS versus QUIC).
+#[derive(Clone, Default)]
+pub(crate) struct ClientExtensionsTemplate<'a> {
+    /// QUIC transport parameters (RFC9001 prior to draft 33)
+    pub(crate) transport_parameters_draft: Option<Payload<'a>>,
+
+    /// QUIC transport parameters (RFC9001)
+    pub(crate) transport_parameters: Option<Payload<'a>>,
+}
```

Comment:
We never have both, right? `Template` seems like an overly generic name for this (at least at this commit).

Reply:
changing this to `ClientExtensionsInput`; grateful for any better ideas